app/Controllers/FrontController.php

```diff
@@ -59,8 +61,9 @@ public function main($table = null, $action = null, $id = null)
         $this->table = $table;
         $this->action = $action;
         $this->id = $id;
-
-        $this->businessModel = new BusinessModel($this->table, $this->action, $this->id);
+        
+        $businessModel = $this->businessModel ;
+        $this->businessModel = new $businessModel($this->table, $this->action, $this->id);
```

Comment:
Yes. For what it's worth, it immediately replaces the only reference to the first one. 

The problem is that the proper BusinessModel needs to have its three arguments set properly. Those are only available in FrontController->main(), when the router passes them as route parameters. I wanted to instantiate a BusinessModel in teh FrontController Constructor so that subclasses of FrontController could override the constructor to easily change the class used for the BusinessModel (like my own subclass of BusinessModel) without having to override FrontController::main(). When you run "new" on an object, it instantiates an object of the same class. Thus, the constructor chooses which class to use, and main() replaces it with a new object injected with the proper constructor arguments.